### PR TITLE
Network area diagram integration

### PR DESCRIPTION
**Pull request by etiennehomer**

### CODE REVIEW

**Comment by SlimaneAmar** [comment]
Need to refresh the NAD when the topoly of the VL change ?

**Comment by SlimaneAmar** [comment]
Also need invalidate the flows when the VL state change

**Review by JacquesBorsenberger — APPROVED** [review]

File: src/components/diagrams/networkAreaDiagram/network-area-diagram.js [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+        sizeWidth = errorWidth; // height is not set so height is auto;
+    } else if (
+        typeof finalPaperWidth != 'undefined' &&
+        typeof finalPaperHeight != 'undefined'
+    ) {
+        sizeWidth = finalPaperWidth;
+        sizeHeight = finalPaperHeight;
+    }
+
+    return !svg.error ? (
```
Comment (laurgarn) [inline src/components/diagrams/networkAreaDiagram/network-area-diagram.js:336]: May be inverse the test to have the shortest branch first ?

Comment (etiennehomer) [reply src/components/diagrams/networkAreaDiagram/network-area-diagram.js:350]: Done

**Review by SlimaneAmar — APPROVED** [review]

**Review by laurgarn — APPROVED** [review]

File: src/components/diagrams/networkAreaDiagram/network-area-diagram.js [hunk 2]
```diff
@@ -0,0 +1,470 @@
...
+                    totalWidth={width}
+                    totalHeight={height}
+                    {...props}
+                />
+            )}
+        </AutoSizer>
+    );
+});
+
+NetworkAreaDiagram.propTypes = {
```
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/components/diagrams/networkAreaDiagram/network-area-diagram.js:457]: it's complete now

**Pull request merged**